### PR TITLE
General cleanup

### PR DESCRIPTION
**Pull request by lenisko**
I went through the whole project and did some cleanup on the way :shrug:

* Replaced deprecated calls
* Handled unhandled errors
* Improved readability of queries (except `incident.go` - you can't escape `` ` `` between back quotes in Go)
* Removed redundant type conversions
* Replaced empty slice declarations using literal
* Ignored inspections on unused variables, functions
* Replaced snake case with camel case for a few variables ~beautiful is better than ugly
* Omitted types in places they are not needed
* Renamed `DbDetails` to `Details` - [go/wiki/CodeReviewComments#package-names](https://github.com/golang/go/wiki/CodeReviewComments#package-names)

It now includes changes from https://github.com/UnownHash/Golbat/pull/36 - sorry, but had ugly merge :)

### CODE REVIEW

**Review by jfberry — COMMENTED** [review]
First of all - **THANKS** - 
Golbat was my first go project and there are a number of rough edges so it's nice to have someone go through the project and give it a spruce up.
With a few changes I'd be happy to get this merged in...

The thing that I want to spend more time on are changes away from empty slices as generally I do this deliberately (and choose to use var when it doesn't matter) - but without spending a bit more time than I have now tracking these down I can't tell whether they matter inside github.  Perhaps you will be able to look more easily.  Certainly some I have highlighted here do cause a functional difference.

File: config/reader.go [hunk 1]
```diff
@@ -3,22 +3,29 @@ package config
...
+//goland:noinspection GoUnusedExportedFunction
 func ReadJsonConfig() {
 	jsonFile, err := os.Open("config.json")
 	// if we os.Open returns an error then handle it
 	if err != nil {
 		panic(err)
 	}
 	// defer the closing of our jsonFile so that we can parse it later on
-	defer jsonFile.Close()
+	defer func(jsonFile *os.File) {
```
[the remark this answers is not in the document]
Comment (lenisko) [reply config/reader.go:19]: It is just checking whatever a file was closed correctly if not it's going to print an error to log. Could be useful to have a thing in a login case something wont go the way it should to whatever reason. Yeah just a wrapper.

File: db/pokestop.go [hunk 2]
```diff
@@ -11,13 +12,28 @@ type QuestLocation struct {
...
-func GetPokestopPositions(db DbDetails, fence geo.Geofence) ([]QuestLocation, error) {
+//goland:noinspection GoUnusedExportedFunction
+func GetPokestopPositions(db Details, fence geo.Geofence) ([]QuestLocation, error) {
 	bbox := fence.GetBoundingBox()
 
-	areas := []QuestLocation{}
-	err := db.GeneralDb.Select(&areas, "SELECT id, lat, lon FROM pokestop "+
-		"WHERE lat > ? and lon > ? and lat < ? and lon < ? and enabled = 1 "+
-		"and ST_CONTAINS(ST_GEOMFROMTEXT('POLYGON(("+fence.ToPolygonString()+"))'), point(lat,lon))",
+	var areas []QuestLocation
```
[the remark this answers is not in the document]
Comment (lenisko) [reply db/pokestop.go:19]: Ah yes, first one would return `null` where second is `[]`. Haven't considered that on marshalling. Will replace them back.

File: geo/geofence.go [hunk 3]
```diff
@@ -16,21 +16,21 @@ type BoundingBox struct {
 	MaximumLongitude float64
 }
 
-func (fence *Geofence) GetBoundingBox() BoundingBox {
```
[the remark this answers is not in the document]
Comment (lenisko) [reply geo/geofence.go:19]: Replaced to the same receiver name as they were different - [go/wiki/CodeReviewComments#receiver-names](https://github.com/golang/go/wiki/CodeReviewComments#receiver-names)

Same goes for below

File: geo/location.go [hunk 4]
```diff
@@ -11,12 +11,15 @@ func (l Location) Tuple() (float64, float64) {
 	return l.Latitude, l.Longitude
 }
 
+//goland:noinspection GoUnusedExportedFunction
```
[the remark this answers is not in the document]
Comment (lenisko) [reply geo/location.go:14]: Haven't removed, just asked linter to... be silent about fact that this method is unused within this project :)

File: tz/main.go [hunk 5]
```diff
@@ -9,18 +9,31 @@ import (
 
 var tzc timezone.Timezonecache
 
-var dbFilename string = "cache/tz"
-var cacheFilename string = "cache/tz-geoJSON.zip"
+var dbFilename = "cache/tz"
```
[the remark this answers is not in the document]
Comment (lenisko) [reply tz/main.go:12]: Just cosmetics. Sure!

File: webhooks/sender.go [hunk 6]
```diff
@@ -42,7 +43,12 @@ func sendWebhooks(queue WebhookQueue) {
 		log.Warningf("Webhook: %s", err)
 		return
 	}
-	defer resp.Body.Close()
+	defer func(Body io.ReadCloser) {
```
[the remark this answers is not in the document]
Comment (lenisko) [reply webhooks/sender.go:46]: As above, up to you though.

File: config/reader.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jfberry) [reply config/reader.go:19]: Closing a file is never going to fail (unless it were not even opened and the handle is invalid). This is just boilerplate for the sake of the linter. Would prefer it to just be nice and simple

File: webhooks/sender.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (jfberry) [reply webhooks/sender.go:46]: Even the go library does not check the error response for resp.Body.Close()

**Comment by lenisko** [comment]
@jfberry corrected all things

**Comment by jfberry** [comment]
This is now quite out of date; are you going to come back to it or should we close for now?

**Comment by lenisko** [comment]
@jfberry I'll go through it a second time when we'll get fewer active PRs. Too complicated to resolve conflicts ;-) I'll close

**Pull request closed without merging**